File: static/extension/discovery/discovery.py
# Copyright (c) 2021, Gluu
#
# Author: Yuriy Z
#
#

from io.jans.model.custom.script.type.discovery import DiscoveryType

class Discovery(DiscoveryType):
    def __init__(self, currentTimeMillis):
        self.currentTimeMillis = currentTimeMillis

    def init(self, customScript, configurationAttributes):
        print "Discovery script. Initializing ..."
        print "Discovery script. Initialized successfully"

        return True

    def destroy(self, configurationAttributes):
        print "Discovery script. Destroying ..."
        print "Discovery script. Destroyed successfully"
        return True

    def getApiVersion(self):
        return 11

    # Returns boolean, true - indicates that script applied changes, false - no modification will be applied to response (script manipulation is ignored)
    # This method is called after discovery response is ready (and passed as `responseAsJsonObject` method parameter). Hence script can override them
    # Note :
    # responseAsJsonObject - is org.json.JSONObject, you can use any method to manipulate JSON
    # context is reference of io.jans.as.server.model.common.ExecutionContext (in https://github.com/JanssenProject/jans-auth-server project, )
    def modifyResponse(self, responseAsJsonObject, context):
        print "Modify discovery response: %s" % responseAsJsonObject

        responseAsJsonObject.accumulate("key_from_script", "value_from_script")

        print "Discovery response modified. Response after modification: %s" % responseAsJsonObject
        return True

File: jans-core/script/src/test/resources/Discovery.java
import io.jans.model.SimpleCustomProperty;
import io.jans.model.custom.script.model.CustomScript;
import io.jans.model.custom.script.type.discovery.DiscoveryType;
import io.jans.service.custom.script.CustomScriptManager;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.json.JSONObject;

import java.util.Map;

/**
 * @author Yuriy Zabrovarnyy
 */
public class Discovery implements DiscoveryType {

    private static final Logger log = LoggerFactory.getLogger(Discovery.class);
    private static final Logger scriptLogger = LoggerFactory.getLogger(CustomScriptManager.class);

    @Override
    public boolean init(Map<String, SimpleCustomProperty> configurationAttributes) {
        log.info("Init of Discovery Java custom script");
        return true;
    }

    @Override
    public boolean init(CustomScript customScript, Map<String, SimpleCustomProperty> configurationAttributes) {
        log.info("Init of Discovery Java custom script");
        return true;
    }

    @Override
    public boolean destroy(Map<String, SimpleCustomProperty> configurationAttributes) {
        log.info("Destroy of Discovery Java custom script");
        return true;
    }

    @Override
    public int getApiVersion() {
        log.info("getApiVersion Discovery Java custom script: 11");
        return 11;
    }

    @Override
    public boolean modifyResponse(Object responseAsJsonObject, Object context) {
        scriptLogger.info("write to script logger");
        JSONObject response = (JSONObject) responseAsJsonObject;
        response.accumulate("key_from_java", "value_from_script_on_java");
        return true;
    }
}
